src/OpenTelemetry/Metrics/MeterProviderSdk.cs

```diff
@@ -412,12 +439,27 @@ internal Batch<Metric> Collect()
 
                     var indexSnapShot = Math.Min(this.metricIndex, MaxMetrics - 1);
                     var target = indexSnapShot + 1;
+                    int metricCountCurrentBatch = 0;
                     for (int i = 0; i < target; i++)
                     {
-                        this.metrics[i].SnapShot();
+                        var metric = this.metrics[i];
+                        if (metric != null)
+                        {
+                            if (metric.InstrumentDisposed)
+                            {
+                                metric.SnapShot();
```

Comment:
nit: `metric.Snapshot()` is always called. Simplify code by only having `this.metrics[i] = null` in the `if` block.

Reply:
The check `metric.InstrumentDisposed` must be done before snapshot all, so not sure how to rearrange the code here..